### PR TITLE
Adding Async dispatcher

### PR DESCRIPTION
**Pull request by nickrandolph**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Feature

### :arrow_heading_down: What is the current behavior?
The existing dispatcher invokes actions to run on the UI thread but it's fire and forget

### :new: What is the new behavior (if this is a feature change)?
The async dispatcher can be used to run an action on the UI thread and await for it to complete

### :boom: Does this PR introduce a breaking change?
No

### :bug: Recommendations for testing
Run all playground samples

### :thinking: Checklist before submitting

- [ ] All projects build
- [ ] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ X ] Rebased onto current develop

### CODE REVIEW

**Comment by nickrandolph** [comment]
Note that this PR introduces the async dispatcher but doe **not** use it within any of the navigation framework.

File: MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            var syncAction = new Action(async () =>
+            {
+                await action();
+                completion.SetResult(true);
+            });
+            RequestMainThreadAction(syncAction, maskExceptions);
+
+            // If we're already on main thread, then the action will
+            // have already completed at this point, so can just return
+            if (completion.Task.IsCompleted)
```
Comment (martijn00) [inline MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs]: Can't we use ValueTask to do this?

Comment (Cheesebaron) [reply MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs]: `TaskCompletionSource` doesn't return a `ValueTask`? What are you proposing?

Comment (martijn00) [reply MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs]: If we use a `ValueTask` we could return the completed status right away if the Task was finished. Not sure if it makes sense to do here.

**Pull request merged**